// reactjs-navbar/dist/index.modern.js
import React, { Component } from 'react';
import uuid from 'react-uuid';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons';

function _inheritsLoose(subClass, superClass) {
  subClass.prototype = Object.create(superClass.prototype);
  subClass.prototype.constructor = subClass;
  subClass.__proto__ = superClass;
}

var style = {"navbar":"_2iAdU","chevron":"_1RBDr","logo":"_R2FsI","rootMenu":"_2trtS","subMenu":"_3D598","rootMenuLink":"_3HDv1","subSubMenu":"_1JYcn","chevronRight":"_1CGNS","chevronBottom":"_1Pj9u","chevronLeft":"_1fX8J","logoLoader":"_2Gsmq","loader":"_3r-ro","help":"_3Kbya"};

var Navbar = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Navbar, _Component);

  function Navbar() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;
    _this.state = {};
    return _this;
  }

  var _proto = Navbar.prototype;

  _proto.shouldRender = function shouldRender(isAuth) {
    if (!isAuth || typeof isAuth === "function" && !isAuth()) {
      return false;
    } else {
      return true;
    }
  };

  _proto.handleOnClick = function handleOnClick(e, onClickFunction) {
    if (e.target === e.currentTarget) {
      onClickFunction();
    }
  };

  _proto.buildSubSubMenu = function buildSubSubMenu(subSubItems) {
    var _this2 = this;

    var markup = [];
    markup = subSubItems.map(function (subSubItem) {
      if (!_this2.shouldRender(subSubItem.isAuth)) {
        return;
      } else {
        return /*#__PURE__*/React.createElement("li", {
          key: uuid(),
          onClick: function onClick(e) {
            return subSubItem.onClick && _this2.handleOnClick(e, subSubItem.onClick);
          }
        }, [
        /*#__PURE__*/
        React.createElement(FontAwesomeIcon, {
          key: uuid(),
          icon: subSubItem.icon
        }),
        /*#__PURE__*/
        React.createElement("a", {
          key: uuid(),
          href: "#"
        }, subSubItem.title), subSubItem.subItems ? /*#__PURE__*/React.createElement("span", {
          key: uuid(),
          className: style.chevron + " " + style.chevronRight
        }) : null, subSubItem.subItems ? /*#__PURE__*/React.createElement("div", {
          key: uuid(),
          className: style.subSubMenu
        }, /*#__PURE__*/React.createElement("ul", null, " ", _this2.buildSubSubMenu(subSubItem.subItems), " ")) : null].map(function (e) {
          return e;
        }));
      }
    });
    return markup;
  };

  _proto.buildMenu = function buildMenu(_ref) {
    var _this3 = this;

    var menuItems = _ref.menuItems;
    return /*#__PURE__*/React.createElement("div", {
      className: style.rootMenu
    }, /*#__PURE__*/React.createElement("ul", null, menuItems.map(function (rootItem) {
      if (!_this3.shouldRender(rootItem.isAuth)) {
        return;
      } else {
        return /*#__PURE__*/React.createElement("li", {
          key: uuid(),
          onClick: function onClick(e) {
            return rootItem.onClick && _this3.handleOnClick(e, rootItem.onClick);
          },
          className: !rootItem.subItems ? style.rootMenuLink : null
        }, [
        /*#__PURE__*/
        React.createElement(FontAwesomeIcon, {
          key: uuid(),
          icon: rootItem.icon,
          size: "lg"
        }),
        /*#__PURE__*/
        React.createElement("a", {
          href: "#",
          key: uuid()
        }, rootItem.title), rootItem.subItems ? /*#__PURE__*/React.createElement("span", {
          key: uuid(),
          className: style.chevron + " " + style.chevronBottom
        }) : null, rootItem.subItems ? /*#__PURE__*/React.createElement("div", {
          key: uuid(),
          className: style.subMenu
        }, /*#__PURE__*/React.createElement("ul", null, rootItem.subItems.map(function (subItem) {
          if (!_this3.shouldRender(subItem.isAuth)) {
            return;
          } else {
            return /*#__PURE__*/React.createElement("li", {
              key: uuid(),
              onClick: function onClick(e) {
                return subItem.onClick && _this3.handleOnClick(e, subItem.onClick);
              }
            }, [
            /*#__PURE__*/
            React.createElement(FontAwesomeIcon, {
              key: uuid(),
              icon: subItem.icon
            }),
            /*#__PURE__*/
            React.createElement("a", {
              key: uuid(),
              href: "#"
            }, subItem.title), subItem.subItems ? /*#__PURE__*/React.createElement("span", {
              key: uuid(),
              className: style.chevron + " " + style.chevronRight
            }) : null, subItem.subItems ? /*#__PURE__*/React.createElement("div", {
              key: uuid(),
              className: style.subSubMenu
            }, /*#__PURE__*/React.createElement("ul", null, _this3.buildSubSubMenu(subItem.subItems))) : null].map(function (e) {
              return e;
            }));
          }
        }))) : null].map(function (e) {
          return e;
        }));
      }
    })));
  };

  _proto.render = function render() {
    var _this4 = this;

    return /*#__PURE__*/React.createElement("div", {
      id: style.navbar
    }, /*#__PURE__*/React.createElement("div", {
      className: style.logoLoader
    }, this.props.logo && /*#__PURE__*/React.createElement("div", {
      className: style.logo
    }, /*#__PURE__*/React.createElement("img", {
      src: this.props.logo,
      alt: "logo"
    })), /*#__PURE__*/React.createElement("div", {
      style: {
        opacity: this.props.isLoading ? "1" : "0"
      }
    }, this.props.loader ? this.props.loader : /*#__PURE__*/React.createElement("div", {
      className: style.loader,
      style: {
        opacity: this.props.isLoading ? "1" : "0"
      }
    }))), this.buildMenu(this.props), this.props.helpCallback && /*#__PURE__*/React.createElement("div", {
      className: style.help,
      onClick: function onClick() {
        return _this4.props.helpCallback();
      }
    }, /*#__PURE__*/React.createElement("a", {
      href: "#"
    }, "Help"), /*#__PURE__*/React.createElement(FontAwesomeIcon, {
      icon: faQuestionCircle,
      size: "lg"
    })));
  };

  return Navbar;
}(Component);

export default Navbar;
